pkg/storage/export/export/export.go

```diff
@@ -1380,32 +1392,6 @@ func (ctrl *VMExportController) pvcConditionFromPVC(pvcs []*corev1.PersistentVol
 	return cond
 }
 
-func (ctrl *VMExportController) expandVirtualMachine(vm *virtv1.VirtualMachine) (*virtv1.VirtualMachine, error) {
-	instancetypeSpec, err := ctrl.instancetypeMethods.FindInstancetypeSpec(vm)
-	if err != nil {
-		return nil, err
-	}
-	preferenceSpec, err := ctrl.instancetypeMethods.FindPreferenceSpec(vm)
-	if err != nil {
-		return nil, err
-	}
-
-	if instancetypeSpec == nil && preferenceSpec == nil {
-		return vm, nil
-	}
-
-	conflicts := ctrl.instancetypeMethods.ApplyToVmi(field.NewPath("spec", "template", "spec"), instancetypeSpec, preferenceSpec, &vm.Spec.Template.Spec, &vm.Spec.Template.ObjectMeta)
-	if len(conflicts) > 0 {
-		return nil, fmt.Errorf("cannot expand instancetype to VM, due to %d conflicts", len(conflicts))
-	}
-
-	// Remove InstancetypeMatcher and PreferenceMatcher, so the returned VM object can be used and not cause a conflict
-	vm.Spec.Instancetype = nil
-	vm.Spec.Preference = nil
```

Comment:
isnt it a problem that you dont perform this lines when you call applytovm instead of this deleted expandVirtualMachine?

Reply:
if so you should add that check to the test I commented about